Fatal error: exception Failure("return gives bool expected int in x")